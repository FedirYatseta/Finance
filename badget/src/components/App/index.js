import React, { Component } from 'react'

import {
    BrowserRouter as Router,
    Routes,
    Route
} from "react-router-dom";
import About from '../About';
import Statistic from '../Statistics';
import Home from '../Home';
import Navbar from '../Nav';
import { open } from '../../utils/indexdb'


class App extends Component {

    constructor(props) {
        super(props)

        this.state = {
            loading: true
        }

    }

    componentDidMount() {
        open().then(() => {
            this.setState({
                loading: false
            })
        }).catch((e) => {
            console.log('Error', e);
        })
    }
    render() {
        if (this.state.loading) {
            return <div> loading</div>
        }
        return (
            <Router>
                <Navbar />
                <Routes>
                    <Route path="/about" element={<About />} />
                    <Route path="/statistics" element={<Statistic />} />
                    <Route path="/" element={<Home />} />
                </Routes>
            </Router>

        )
    }

}

export default App;