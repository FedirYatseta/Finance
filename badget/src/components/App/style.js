import styled from 'styled-components'


export const Li = styled.li`
        display: flex;
        align-item: center;
        justify-content: center;
        border-radius: 4px;
        padding: 5px;
        margin: 5px 10px;
        font-size:32px;
        a{
                text-decoration: none;
        }
`;

export const Nav = styled.nav`
        display: flex;
        align-item: center;
        justify-content: center;
        text-decoration: none;
`;

export const Ul = styled.ul`
        display: flex;
        align-item: center;
        justify-content: center;
        border-radius: 4px;
        padding: 5px;
        margin: 5px 10px;
        font-size:32px;
        text-decoration: none;
`;





