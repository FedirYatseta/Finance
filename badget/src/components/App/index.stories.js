import React from 'react';

import App from '.';
import { Navbar } from '../Nav';

export default {
    title: 'Example/App',
    component: App,
};

const Template = (args) => <App {...args} />;

export const Income = Template.bind({});
Income.args = {


};