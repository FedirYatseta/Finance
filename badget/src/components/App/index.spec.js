

//import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '.';



describe('App', () => {
  it('should show App', () => {
    const component = renderer.create(
      <App />,
    );
    expect(component).toMatchSnapshot();

  })
});