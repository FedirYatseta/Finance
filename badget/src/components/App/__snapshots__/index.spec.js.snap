// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should show App 1`] = `
<div>
   loading
</div>
`;

exports[`Transaction should show Transaction 1`] = `
<div>
   loading
</div>
`;
